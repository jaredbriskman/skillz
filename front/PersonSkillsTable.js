import { PersonSkillRow, personSkillsBySkill } from './Person.js'
import React, { Component } from 'react';
import { graphql, withApollo } from 'react-apollo';

import gql from 'graphql-tag'

class PersonSkillTable extends Component {
    render() {
        let { data } = this.props;
        if (!data.allUsers) return null;
        let people = data.allUsers.edges.map(({ node }) => node);
        let skills = data.allSkills.edges.map(({ node }) => node);
        people.sort(({ firstName: a }, { firstName: b }) => a < b ? -1 : a > b ? 1 : 0)
        return <table className="ui striped selectable definition table">
            <tbody>
                <tr><th />{skills.map(({ name, id }) => <th key={'skill-' + id}>{name}</th>)}</tr>
                {people.map(person =>
                    <PersonSkillRow key={'person-' + person.id}
                        person={person}
                        skills={personSkillsBySkill(person, skills)}
                        onClick={_ => this.props.onRowClick(person)}
                    />)}
            </tbody>
        </table>
    }
}

export const personSkillsQuery = gql`
query {
  allUsers {
      edges {
        node {
          id
          firstName
          lastName
          userSkillsByUserId {
            edges {
              node {
                id
                skillId
                experience
                desire
                note
                instructor
                skillBySkillId {name}
              }
            }
          }
        }
      }
    }
    allSkills(orderBy: NAME_ASC) {
      edges {
        node {
          id
          name
        }
      }
    }
}`;


export const PersonSkillTableContainer = graphql(personSkillsQuery)(PersonSkillTable);
