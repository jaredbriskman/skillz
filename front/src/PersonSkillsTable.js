import { PersonSkillRow, personSkillsBySkill } from './Person.js'
import React, { Component } from 'react';
import { graphql, withApollo } from 'react-apollo';

import gql from 'graphql-tag'

export const PersonSkillsTable = ({ data }) => {
  if (!data.allUsers) return null;
  let people = data.allUsers.edges.map(({ node }) => node);
  let skills = data.allSkills.edges.map(({ node }) => node);
  people.sort(({ firstName: a }, { firstName: b }) => a < b ? -1 : a > b ? 1 : 0)
  return <table className="ui striped selectable definition table">
    <tbody>
      <tr><th />{skills.map(({ name, id }) => <th key={id}>{name}</th>)}</tr>
      {people.map(person =>
        <PersonSkillRow key={person.id}
          person={person}
          skills={personSkillsBySkill(person, skills)}
          onClick={() => this.props.onRowClick(person)}
        />)}
    </tbody>
  </table>
}

export const personSkillsQuery = gql`
query {
  allUsers {
      edges {
        node {
          id
          firstName
          lastName
          userSkillsByUserId {
            edges {
              node {
                id
                skillId
                experience
                desire
                skillBySkillId {name}
              }
            }
          }
        }
      }
    }
    allSkills(orderBy: NAME_ASC) {
      edges {
        node {
          id
          name
        }
      }
    }
}`;


export const PersonSkillTableContainer = graphql(personSkillsQuery)(PersonSkillsTable);
