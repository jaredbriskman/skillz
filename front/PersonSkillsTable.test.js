import { PersonSkillTable } from './PersonSkillsTable.js'
import React from 'react'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const personSkillsData = {
    allUsers: {
        edges: [
            {
                node: {
                    id: 1,
                    firstName: "Laszlo",
                    lastName: "Toth",
                    userSkillsByUserId: {
                        edges: [
                            {
                                node: {
                                    id: 3,
                                    skillId: 1,
                                    experience: 5,
                                    desire: 3,
                                    note: null,
                                    skillBySkillId: {
                                        name: "React"
                                    }
                                }
                            }
                        ]
                    }
                }
            }
        ]
    },
    allSkills: {
        edges: [
            {
                node: {
                    id: 13,
                    name: "React"
                }
            },
            {
                node: {
                    id: 7,
                    name: "Jest"
                }
            }
        ]
    }
};

describe('PersonSkillTable', () => {
    test(`matches snapshot`, () => {
        const component = renderer.create(
            <PersonSkillTable data={personSkillsData} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
