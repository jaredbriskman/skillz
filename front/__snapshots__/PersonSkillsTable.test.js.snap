// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonSkillTable matches snapshot 1`] = `
<table
  className="ui striped selectable definition table"
>
  <tbody>
    <tr>
      <th />
      <th>
        React
      </th>
      <th>
        Jest
      </th>
    </tr>
    <tr
      onClick={[Function]}
    >
      <th>
        <div>
          Laszlo
        </div>
        <div
          className="right aligned"
        >
          Toth
        </div>
      </th>
      <td>
        <div
          className="compact my-rating"
        />
        <div
          className="compact my-rating"
        />
      </td>
      <td>
        <div
          className="compact my-rating"
        />
        <div
          className="compact my-rating"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
